### PR TITLE
Form

### PR DESCRIPTION
**Pull request by farahccino**
Happy Monday, everyone 🙂 I've embedded my form into my home page. Please test this:
- [x] go to plus page and add some daily activities (only choose täglich)
- [x] go to the home page (also called today page) and edit your activities
- [x] edit mode should open the form as modal 
- [x] you can edit your activities or click on the back button to abort

Thanks a lot.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+# Plan Your activities with Cherish!
+
+This innovative app gives You an overview of Your activities and Your mood. Thanks to the tracker You can reflect upon Your week. Cherish has an intuitive design and serves as a reminder, tracker and journal.
+
+# Plane Deine Aktivitäten mit Cherish!
+
+Mit dieser innovativen App behältst Du den Überblick über Deine Aktivitäten und Deine Stimmung. Dank des Trackers kannst Du reflektieren, wie Deine Woche verlaufen ist. Cherish hat ein intuitives Design und dient als Reminder, Tracker und Journal.
+
 # Getting Started with Create React App
```
Comment (SabMueller) [inline README.md:9]: I really like your Readme! It gives an excellent overview. :)

File: src/components/HabitForm.js [hunk 2]
```diff
@@ -1,72 +1,78 @@
...
-import styled from 'styled-components/macro';
-
+import PropTypes from "prop-types";
+import { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+import { v4 as uuidv4 } from "uuid";
 
+import styled from "styled-components/macro";
 
 HabitForm.propTypes = {
```
Comment (SabMueller) [inline src/components/HabitForm.js:8]: You've used propTypes - awesome! :)

File: src/components/HabitForm.js [hunk 3]
```diff
@@ -1,72 +1,78 @@
...
-  useEffect(() => {
-    window.localStorage.setItem('habit', JSON.stringify(habit))
-  }, [habit]);
 
-  function addHabit(habit) {
-    setHabit([...habit, habit]);
-  }
+  let history = useHistory();
+  const goToPreviousPath = () => {
+    history.push("/today");
```
Comment (SabMueller) [inline src/components/HabitForm.js:56]: Very cool and convenient! :)

**Review by SabMueller — APPROVED** [review]
Looks good to me! Very organized and clean code! Good job! ✨ :)

File: src/lib/localStorage.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export function updateLocalStorage(key, data) {
+    localStorage.setItem(key, JSON.stringify(data));
+  }
+  
+  export function loadFromLocalStorage(key) {
+    try {
+      const localData = localStorage.getItem(key);
+      return JSON.parse(localData);
+    } catch (error) {
+      console.error(error);
```
Comment (hugh-burgess) [inline src/lib/localStorage.js:10]: You may also like:

```
export function getItemsFromLocalStorage(key) {

          const data = JSON.parse(localStorage.getItem(key)) || [];
          return data;
 ```

File: src/App.js [hunk 5]
```diff
@@ -1,47 +1,84 @@
...
+  const [activePage, setActivePage] = useState("today");
+  const [habitToEdit, setHabitToEdit] = useState(null);
+
+  const [habits, setHabits] = useState(
+    loadFromLocalStorage("cherishHabits") ?? []
+  );
+  console.log(habits);
+  useEffect(() => {
+    updateLocalStorage("cherishHabits", habits);
+  }, [habits]);
```
Comment (hugh-burgess) [inline src/App.js:24]: This could go in a `hooks` folder to save you some space.

File: src/App.js [hunk 6]
```diff
@@ -1,47 +1,84 @@
...
-          <Plus />
-        </Route>
-        <Route path='/goals'>
-          <Goals />
-        </Route>
-        </Switch>
-    </MainContainer>
-    </div>
+      <div>
+        <Headline></Headline>
```
Comment (hugh-burgess) [inline src/App.js:44]: Why not just `<Headline/>` ?

File: src/components/HabitForm.js [hunk 7]
```diff
@@ -1,72 +1,78 @@
...
 
   return (
     <Form onSubmit={handleFormSubmit}>
       <label htmlFor="ZielName">Ziel</label>
       <Ziel
         type="text"
         name="ziel"
-        onChange={updateField}
-        value={fieldValue}
+        onChange={updateHabit}
```
Comment (hugh-burgess) [inline src/components/HabitForm.js:65]: For slightly more specificity for future you, why not try this as `handleUpdateHabitChange`

File: src/pages/Home.js [hunk 8]
```diff
@@ -1,71 +1,164 @@
...
+
+const HabitWrapper = styled.div`
+  background-color: rgba(255, 255, 255, 0.7);
+  backdrop-filter: blur(10px);
+  display: block;
+  position: absolute;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
```
Comment (hugh-burgess) [inline src/pages/Home.js:154]: is this needed? What are you wanting to create here?

**Review by hugh-burgess — COMMENTED** [review]
Great work! Couple things that may help you out.

File: src/lib/localStorage.js (hunk 4 above)
Comment (farahccino) [reply src/lib/localStorage.js:10]: I like this version better!

File: src/App.js (hunk 6 above)
Comment (farahccino) [reply src/App.js:44]: True, I've adjusted it now.

File: src/pages/Home.js (hunk 8 above)
Comment (farahccino) [reply src/pages/Home.js:154]: It is, I created a modal and I want it to be placed in the middle of the screen.

File: src/App.js (hunk 5 above)
Comment (farahccino) [reply src/App.js:24]: 🤔

File: src/pages/Home.js (hunk 8 above)
Comment (hugh-burgess) [reply src/pages/Home.js:154]: does `margin: 0, auto;` work?

https://stackoverflow.com/questions/3170772/what-does-auto-do-in-margin0-auto

File: src/App.js (hunk 5 above)
Comment (hugh-burgess) [reply src/App.js:24]: to create a customHook instead?

File: src/pages/Home.js (hunk 8 above)
Comment (farahccino) [reply src/pages/Home.js:154]: for some reason it doesn't work here. the modal gets placed into the bottom left corner.

File: src/App.js (hunk 5 above)
Comment (farahccino) [reply src/App.js:24]: I'll look into that after some progress on my app.

**Pull request merged**